squad/core/comparison.py

```diff
@@ -108,37 +108,66 @@ def diff(self):
         return self.__diff__
 
     __regressions__ = None
+    __progressions__ = None
 
     @property
     def regressions(self):
-        if self.__regressions__ is not None:
+        return self.__regressions_progression__()
+
+    @property
+    def progressions(self):
+        return self.__regressions_progression__(False)
+
+    def __regressions_progression__(self, regression=True):
+        comparison_tuple = ('pass', 'fail')
+        if not regression:
+            comparison_tuple = ('fail', 'pass')
+        if regression and self.__regressions__ is not None:
             return self.__regressions__
 
+        if not regression and self.__progressions__ is not None:
+            return self.__progressions__
+
         if len(self.builds) < 2:
             self.__regressions__ = {}
+            self.__progressions__ = {}
+            if not regression:
+                return self.__progressions__
             return self.__regressions__
 
-        regressions = OrderedDict()
+        comparisons = OrderedDict()
         after = self.builds[-1]  # last
         before = self.builds[-2]  # second to last
         for env in self.environments[after]:
-            regression_list = []
+            comparison_list = []
             for test, results in self.diff.items():
                 results_after = results.get((after, env))
                 results_before = results.get((before, env))
-                if (results_before, results_after) == ('pass', 'fail'):
-                    regression_list.append(test)
-            if regression_list:
-                regressions[env] = regression_list
-
-        self.__regressions__ = regressions
+                if (results_before, results_after) == comparison_tuple:
+                    comparison_list.append(test)
+            if comparison_list:
+                comparisons[env] = comparison_list
+
+        if not regression:
+            self.__progressions__ = comparisons
+            return self.__progressions__
+        self.__regressions__ = comparisons
```

Comment:
I think this method should just return the list of status changes, and leave the assignment to the caller method